auto_submit/lib/service/validation_service.dart

```diff
@@ -334,8 +335,13 @@ Exception: ${exception.message}
     final github.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
     final int number = messagePullRequest.number!;
     // Pass an explicit commit message from the PR title otherwise the GitHub API will use the first commit message.
-    String commitMessage = messagePullRequest.title!;
-    commitMessage = commitMessage.replaceFirst('Revert "Revert', 'Reland'); // Cleanup auto-generated revert messages.
+    final String prTitle =
+        messagePullRequest.title!.replaceFirst('Revert "Revert', 'Reland'); // Cleanup auto-generated revert messages.
```

Comment:
i understand this is not really part of your PR so feel free to ignore, but:

fwiw i would avoid "Reland" as well (also if i'm not misreading this code it'll cause a trailing quote mark).
would be ideal if the commit message was just the original one and the pr/commit description mentioned that it was a reland (ideally with a link to the revert PR).

Reply:
For context, I rely on our commit log daily to find good branch points without having to wait hours for tests.

Relands indicate to me that a commit is risky to rely on, and it saves me time to  avoid including it in the branch. Unfortunately, I need to read ~80 commits a day, so I'd prefer not to read the whole message.

Reply:
So, I think I found a compromise:

The initial change that introduced this functionality is actually what regressed https://github.com/flutter/flutter/issues/124554. It also introduced the weird behavior where the commit now has the title twice (GitHub was still reading the PR title, appending the PR number, and setting that as the first line of the commit message, since github.dart is passing a null commit_title to the API).

Also, it seems the string `Revert "Revert` appears relatively infrequently in our PR titles (this was the only example I can find across framework and engine that appeared to actually use the substitution https://github.com/flutter/engine/commit/56bebec98b81).

So now, we only include the title in the message IFF the title matches the `Revert "Revert` pattern. (If we ever want to really set the title for every merge commit, I think we'll need to upstream a change to github.dart to set the `commit_title` field in the JSON we send to the github API).